### PR TITLE
List instruments cu 86b29nc6d

### PR DESCRIPTION
**Pull request by tianning-zhang**

### CODE REVIEW

**Comment by platform-automation-dnastack** [comment]
Task linked: [CU-86b29nc6d Add CLI command to list instruments](https://app.clickup.com/t/86b29nc6d)

File: tests/cli/test_workbench.py [hunk 1]
```diff
@@ -1382,4 +1382,28 @@ def test_workflow_transformation_delete(self):
...
             parse_output=False
         )
-        self.assertTrue("Deleted..." in output)
\ No newline at end of file
+        self.assertTrue("Deleted..." in output)
+
+    def test_instruments_list(self):
+        created_storage_account = self._create_storage_account()
+        created_platform = self._create_platform(created_storage_account)
+        instruments = self._wait_for_instruments()
```
Comment (patmagee) [inline tests/cli/test_workbench.py:1390]: this line is effectively testing the same thing as below. I suggest only calling the `self._wait()`

**Review by patmagee — COMMENTED** [review]
@tianning-zhang just a small change in regards to the wait that is happening in the test.

File: tests/cli/test_workbench.py (hunk 1 above)
Comment (tianning-zhang) [reply tests/cli/test_workbench.py:1390]: @patmagee do you mean like this?
```
def test_instruments_list(self):
        created_storage_account = self._create_storage_account()
        created_platform = self._create_platform(created_storage_account)
        self._wait()
        instruments = [Instrument(**instrument) for instrument in self.simple_invoke(
            'workbench', 'instruments', 'list'
        )]
        self.assert_not_empty(instruments, f'Expected at least one instrument. Found {instruments}')
        for instrument in instruments:
            self.assert_not_empty(instrument.id, 'Instrument ID should not be empty')
```

**Review by patmagee — APPROVED** [review]
LGTM

**Pull request merged**